### PR TITLE
Wax golem rite and new bible effects for candle sect and kudzu sect

### PR DESCRIPTION
**Pull request by Marmio64**
# Document the changes in your pull request

Adds a rite to the candle sect allowing them to transform people into wax golems. Wax golems are similar to cloth golems in that they're flammable and can revive after death when in the proper condition, but are also very different. Instead of being vulnerable to fire they're very resistant to it (but still take fire damage unlike other golems) and if they die while on fire they'll melt into wax, and then revive after about 80 seconds unless the wax pile is destroyed. Dying while not on fire will not result in them melting and reviving after short while. Their other stats are: a speed mod of 1, lower punch damage, 10 armor, and 0.75 burn mod. Wax golem is designed as an alternative to fireproofing your clothes.

waxgolem
codersprite wax golem with his dumb candle head and half melted body

Also adds the ability to heal burn damage of people (only if they are on fire) for candle sect, and allows plant sect to heal people who are in the plants or vines faction.

# Wiki Documentation

see above

# Changelog

:cl:  
rscadd: Adds wax golem and the ability to transform people into it for the candle sect. Also adds new candle sect and kudzu sect bless effects.
/:cl:

### CODE REVIEW

**Comment by ChesterTheCheesy** [comment]
is this dead

**Comment by Marmio64** [comment]
been busy but i'd like to fix it soon

**Review by nesisknesis — APPROVED** [review]

**Comment by bruhlookatthisdood** [comment]
*cutely lights the Chaplin on fire with a lighter*

**Comment by ToasterBiome** [comment]
@Marmio64

**Comment by JamieD1** [comment]
Fix conflicts please

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -1387,3 +1387,81 @@
...
+/datum/species/golem/wax/spec_death(gibbed, mob/living/carbon/human/H)
+	if(gibbed)
+		return
+	if(H.on_fire)
+		H.visible_message(span_danger("[H] melts apart into wax!"))
+		new /obj/structure/wax_pile(get_turf(H), H)
+		return
+	else
+		return
+	..()
```
Comment (ToasterBiome) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion

```

unreachable code, make sure this does what you want.

**Review by ToasterBiome — CHANGES_REQUESTED** [review]

**Comment by Marmio64** [comment]
> Fix conflicts please

i believe its fixed

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
@Marmio64 it is now conflicting again

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -1387,3 +1387,80 @@
...
+
+/obj/structure/wax_pile
+	name = "pile of wax"
+	desc = "It emits a strange aura, as if there was still life within it..."
+	max_integrity = 150
+	armor = list("melee" = 50, "bullet" = 50, "laser" = 25, "energy" = 80, "bomb" = 50, "bio" = 100, "fire" = 95, "acid" = -50)
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "pile_wax"
+
+	var/revive_time = 800
```
Comment (ToasterBiome) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```suggestion
	var/revive_time = 80 SECONDS
```

File: code/modules/religion/religion_sects.dm [hunk 3]
```diff
@@ -277,8 +305,33 @@
...
+		var/obj/item/bodypart/BP = X
+		if(BP.status == BODYPART_ROBOTIC)
+			to_chat(user, span_warning("[GLOB.deity] refuses to heal this metallic taint!"))
+			return 0
+
+	var/heal_amt = 10
+	var/list/hurt_limbs = H.get_damaged_bodyparts(1, 1, null, BODYPART_ORGANIC)
+
+	if(hurt_limbs.len)
+		for(var/X in hurt_limbs)
```
Comment (ToasterBiome) [inline code/modules/religion/religion_sects.dm]:
```suggestion
		for(var/obj/item/bodypart/affecting in hurt_limbs)
```

File: code/modules/religion/religion_sects.dm [hunk 4]
```diff
@@ -277,8 +305,33 @@
...
+			var/obj/item/bodypart/affecting = X
```
Comment (ToasterBiome) [inline code/modules/religion/religion_sects.dm]:
```suggestion
```

File: code/modules/religion/religion_sects.dm [hunk 5]
```diff
@@ -277,8 +305,33 @@
 	rites_list = list(/datum/religion_rites/plantconversion, /datum/religion_rites/photogeist)
 	altar_icon_state = "convertaltar-green"
 
-//plant sect bibles don't heal or do anything special apart from the standard holy water blessings
-/datum/religion_sect/plant/sect_bless(mob/living/blessed, mob/living/user)
+//plant sect bibles will only heal plant-like things
+/datum/religion_sect/plant/sect_bless(mob/living/L, mob/living/user)
```
Comment (ToasterBiome) [inline code/modules/religion/religion_sects.dm:309]: this heals every damaged limb by 10, is this intended? or just one limb?

**Review by ToasterBiome — DISMISSED** [review]

File: code/modules/religion/religion_sects.dm (hunk 5 above)
Comment (Marmio64) [reply code/modules/religion/religion_sects.dm:309]: Either works tbh but it was supposed to be one limb

Comment (ToasterBiome) [reply code/modules/religion/religion_sects.dm:309]: > 

fix

Comment (Marmio64) [reply code/modules/religion/religion_sects.dm:309]: ive changed my mind its now intended and no longer requires me to put more effort into this PR

**Pull request merged**